src/spec/models/session_spec.rb

```diff
@@ -84,6 +84,71 @@
     assert session.errors[:timeslot_id]
   end
 
+  it 'should allow rooms to be swapped' do
+    room1 = create(:room)
+    room2 = create(:room)
+    slot1 = create(:timeslot_1)
+
+    Session.new(:title => 'Session 1', :description => 'First session').tap do |s|
+      s.timeslot = slot1
+      s.room = room1
+      s.participant = joe
+      s.event = event
+      s.save!
+    end
+    session1 = Session.last
+    
+    Session.new(:title => 'Session 2', :description => 'Second session').tap do |s|
+      s.timeslot = slot1
+      s.room = room2
+      s.participant = luke
+      s.event = event
+      s.save!
+    end
+    session2 = Session.last
+
+    assert_equal(session1.room, room1)
```

Comment:
This works but I generally prefer the `expects` syntax for new code. Having said that you might have your expected and actual parameters flipped. (expected=room1, actual=session1.room) I think.